scr/main/java/bittercode/util/DatabaseConfig.java

```diff
@@ -0,0 +1,30 @@
+package com.bittercode.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+class DatabaseConfig {
+
+    static Properties prop = new Properties();
+    static {
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        InputStream input = classLoader.getResourceAsStream("application.properties");

```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider closing InputStream after use</b></div>
<div id="fix">

Consider closing the `InputStream` after use to prevent resource leaks. You can use a try-with-resources statement to ensure the stream is closed automatically.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -12,8 +12,8 @@
          ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
 -        InputStream input = classLoader.getResourceAsStream("application.properties");
 -
 -        try {
 -            prop.load(input);
 +        try (InputStream input = classLoader.getResourceAsStream("application.properties")) {
 +            if (input != null) {
 +                prop.load(input);
 +            }
          } catch (IOException e) {
              e.printStackTrace();
          }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/3#issuecomment-2434572244>#7fa0a4</a></i></small>
</div>


---
🚩 Problem result:
- [ ] This issue was flagged incorrectly